android-client-sdk/src/main/java/com/devcycle/sdk/android/eventsource/SafeBackgroundEventSource.kt

```diff
@@ -0,0 +1,56 @@
+package com.devcycle.sdk.android.eventsource
+
+import com.devcycle.sdk.android.util.DevCycleLogger
+import com.launchdarkly.eventsource.EventSource
+import com.launchdarkly.eventsource.background.BackgroundEventSource
+import java.util.concurrent.RejectedExecutionException
+
+/**
+ * A safe wrapper around BackgroundEventSource that handles RejectedExecutionException
+ * during shutdown operations gracefully to prevent app crashes.
+ * 
+ * This wrapper is necessary because when EventSource is closed, its internal executor
+ * is shut down. If there are any pending tasks or race conditions during shutdown,
+ * they would throw RejectedExecutionException which could crash the application.
+ */
+class SafeBackgroundEventSource(
```

Comment:
Were you able to reproduce the issue that was seen? Are you sure the Event Source doesn't handle this internally? Should we be making changes to the EventSource library instead?